### ISSUE TITLE
Add asserts on top of file - before imports

### ISSUE THREAD

**Issue by AbdealiJK** [timothycrosley/isort#393]
Hi,

Im trying to use `isort`  in [coala](https://github.com/coala-analyzer/coala) and I came across a problem.
In our `setup.py` we do:

```
from coalib import assert_supported_version
assert_supported_version()

# Now begin all other imports
import locale
import sys
```

This is because we do not want users to run our script with python2 - as it throws syntax errors. Now, the assertion has to come before imports as otherwise the imports can throw syntax errors.

How would I go about acheiving this with `isort` ? Is there an arg I can use ? (I couldnt find anything obvious I could use from [here](https://github.com/timothycrosley/isort/wiki/isort-Settings) )

**Comment by timothycrosley** [comment]
No this is not supported yet, but is something I would like to support.

**Comment by timothycrosley** [comment]
Thanks for raising the issue :)

**Comment by jayvdb** [comment]
> This is because we do not want users to run our script with python2 - as it throws syntax errors.

If the first block of imports is std lib imports, there will not be any syntax errors.
But there may be import errors, as some of the names of std lib imports has changed.

[Pywikibot]([https://github.com/wikimedia/pywikibot-core) needs something similar, as we have a `pywikibot.tools.PY2` bool that we import very early in the import sequence so that it can be used for our Python 2/3 conditional imports, including std lib.

So, it would be great to have a list of names that can be imported and used before std lib imports.

**Comment by bootandy** [comment]
I have a solution to this but I'm not sure if its very clean. It will fix the above issue though. 

This commit would allow you to add the comment:  "# isort:split-import-group" next to the assert and then isort would treat the imports below and above the comment as separate groups to sort. 

https://github.com/bootandy/isort/commit/5f3a0ff843bf8c8ab6ffe962e77738bf03dd6429

Thoughts? Worth merging or is there a cleaner way to do this?

 @timothycrosley

**Comment by andersk** [comment]
I’m also interested in this functionality, for code that needs to modify `sys.path` before other `import` statements. This kind of pattern needs _three_ groups:

```python
import site
site.addsitedir(…)  # isort:split-import-group

__requires__ = ["foo", "bar", …]
import pkg_resources  # isort:split-import-group

import foo
import bar
…
```

**Comment by timothycrosley** [comment]
isort support this in the latest development branch, to be released shortly as the 5.0.0 release.

Syntax is:

```
import site
site.addsitedir(…)
__requires__ = ["foo", "bar", …]  # isort:split-import-group

# isort:split
import pkg_resources 

import foo
import bar
```
